indico/modules/events/timetable/client/js/EntryPopup.tsx

```diff
@@ -94,6 +94,45 @@ function TimetablePopupContent({
     dispatch(actions.setDraftEntry(draftEntry));
   };
 
+  const onDelete = async (e: MouseEvent) => {
+    if (!draftEntry.id) {
+      return;
+    }
+
+    e.stopPropagation();
+
+    const deleteURL = {
+      [EntryType.Break]: breakURL({event_id: eventId, break_id: draftEntry.id}),
+      [EntryType.SessionBlock]: sessionBlockURL({
+        event_id: eventId,
+        session_block_id: draftEntry.id,
+      }),
+      [EntryType.Contribution]: contributionURL({event_id: eventId, contrib_id: draftEntry.id}),
+    }[draftEntry.type];
+
+    try {
+      await indicoAxios.delete(deleteURL);
+
+      const deleteHandlers = {
+        [EntryType.Break]: () => dispatch(actions.deleteBreak(draftEntry)),
+        [EntryType.SessionBlock]: () => dispatch(actions.deleteBlock(draftEntry.id)),
+        [EntryType.Contribution]: () =>
+          dispatch(actions.unscheduleEntry(draftEntry as ContribEntry)),
+      };
+
+      const deleteHandler = deleteHandlers[draftEntry.type];
+
+      if (!deleteHandler) {
+        throw new Error('Invalid entry type');
+      }
+
+      deleteHandler();
```

Comment:
I think this should be awaited so we don't trigger onClose until the request actually finished.

I would also expect the error handler below to not trigger atm, because you exited the try block by the time the request's response actually arrives.